File: portal/backend/app/admin/api/router.py
# -*- coding: utf-8 -*-
"""后台管理路由装配入口"""


def register_admin_routes(bp, db, models):
    """初始化后台管理路由"""
    from backend.app.admin.api.auth import init_auth_api
    from backend.app.admin.api.logs import init_logs_api
    from backend.app.admin.api.menu import init_menus_api
    from backend.app.admin.api.users import init_users_api
    from backend.app.admin.api.roles import init_roles_api
    from backend.app.admin.api.dicts import init_dicts_api
    from backend.app.admin.api.scheduled_task import init_scheduled_task_api
    from backend.app.admin.api.dashboard import init_dashboard_api
    from backend.app.admin.api.notification import init_notification_api
    from backend.app.admin.api.announcement import init_announcement_api
    from backend.app.admin.api.settings import init_settings_api

    init_auth_api(bp, db, models)
    init_users_api(bp, db, models)
    init_roles_api(bp, db, models)
    init_menus_api(bp, db, models)
    init_logs_api(bp, db, models)
    init_dicts_api(bp, db, models)
    init_scheduled_task_api(bp, db, models)
    init_dashboard_api(bp, db, models)
    init_notification_api(bp, db, models)
    init_announcement_api(bp, db, models)
    init_settings_api(bp, db, models)

File: apps/api/src/modules/admin/router.ts
/**
 * admin domain router assembly
 */

import type { FastifyInstance } from 'fastify'
import { registerAnnouncementRoutes } from './announcement/routes'
import { registerAuthRoutes } from './auth/routes'
import { registerDashboardRoutes } from './dashboard/routes'
import { registerDictRoutes } from './dicts/routes'
import { registerLogsRoutes } from './logs/routes'
import { registerMenuRoutes } from './menu/routes'
import { registerNotificationRoutes } from './notification/routes'
import { registerRoleRoutes } from './roles/routes'
import { registerScheduledTaskRoutes } from './scheduled-task/routes'
import { registerUserRoutes } from './users/routes'

export async function registerAdminRoutes(app: FastifyInstance): Promise<void> {
  // Register logs first: it installs the global onResponse audit hook
  await registerLogsRoutes(app)
  await registerAuthRoutes(app)
  await registerUserRoutes(app)
  await registerRoleRoutes(app)
  await registerMenuRoutes(app)
  await registerDictRoutes(app)
  await registerScheduledTaskRoutes(app)
  await registerDashboardRoutes(app)
  await registerNotificationRoutes(app)
  await registerAnnouncementRoutes(app)
}
